test/extended/node_tuning/node_tuning.go

```diff
@@ -111,6 +111,77 @@ var _ = g.Describe("[sig-node-tuning] NTO should", func() {
 
 		err = fmt.Errorf("case: %v\nexpected error got because of %v", g.CurrentSpecReport().FullText(), fmt.Sprintf("stalld service restarted : %v", errWait))
 		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	//OCPBUGS-18052
+	g.It("SNO installation does not finish due to wait for non-existing machine-config [Early]", func() {
+		isSNO := isSNOCluster()
+		if !isSNO {
+			g.Skip("only test on SNO cluster, skipping it ...")
+		}
+
+		firstMasterNodeNameStr := getFirstMasterNode()
+		firstMasterNodeName := strings.Trim(firstMasterNodeNameStr, "'")
+
+		e2e.Logf("assert the status of mcp master")
+		// Wait for 10 minutes and check mcp master stay on normal state
+		errWait := wait.Poll(30*time.Second, 600*time.Second, func() (bool, error) {
+			mcpUpdatingStatus, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("mcp", "master", `-ojsonpath='{.status.conditions[?(@.type=="Updating")].status}'`).Output()
```

Comment:
I believe oc.AsAdmin() should be used as a last resort.  Is it possible to have a mcpClient in a similar way as nodeClient below?  Similarly below. Then you wouldn't need any string manipulation functions (Contains/Trim/etc.) and you'd only query object attributes.